File: DSA/50_Days_50_Questions/209-minimum-size-subarray-sum/minimum-size-subarray-sum.py
class Solution(object):
    def minSubArrayLen(self, target, nums):
        n = len(nums)
        ans = float('inf')
        l = 0
        sum = 0

        for r in range(n):
            sum += nums[r]

            while (sum>=target):
                win_len = r - l + 1
                ans = min (ans, win_len)

                sum -=nums[l]
                l +=1
        if(ans == float('inf')):
            return 0
        else:
            return ans 

File: DSA/50_Days_50_Questions/209-minimum-size-subarray-sum/minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int n = nums.size();
        int sum  = 0;
        int ans = INT_MAX;
        int left = 0;

        for(int r=0; r<n; r++){
            sum += nums[r];

            while (sum>=target){
                int win_len = r - left + 1;
                ans = min(ans , win_len);

                sum -= nums[left];
                left ++;
            }
        }
         return (ans == INT_MAX) ? 0 : ans; 

        
    }
};